hare/messagevalidation.go

```diff
@@ -36,7 +36,14 @@ func (validator *MessageValidator) ValidateMessage(m *pb.HareMessage, k uint32)
 	}
 
 	// validate signature
-	if !validator.signing.Validate(data, m.InnerSig) {
+	pk, err := crypto.NewPublicKey(m.PubKey)
+	if err != nil {
+		log.Error("Validate message failed: could'nt parse pubkey")
+		return false
+	}
+
+	ok, err := pk.Verify(data, m.InnerSig)
```

Comment:
Why not pass the pubKey to the Signing instance?
We can keep the interface as is and internally, use pk for validation for example.
I think it will be cleaner as we will not need to handle the same errors over and over again.

Reply:
for every message you need to verify with a different pubkey. we'll have to extract the pubkey from the message.
the only way we can use `Signing` is write a function like this `func Verify(pk crypto.PublicKey, sign, msg []byte) (bool, error) { return pk.Verify(sign, msg) }`